Drop support for EOL Ruby versions (`2.5` and `2.6`)

nickmendezFlatiron opened the pull request:
### Summary

I Completed the 3 tasks in Issue #2529
- removed version checks in test_faker_birthday_in_leap_year.rb and test_faker_id_number.rb
- Updated Targeted Ruby Version for Rubocop to 2.7
- Updated spec.required_ruby_version = '>= 2.5' to 2.7

### Other Information

<!-- If there's anything else that's important and relevant to your pull request, mention that information here.

If you are creating a new generator, share how you are going to use it. Use cases are important to make sure that our faker generators are useful. Also, add `@faker.version next` to help users know when a generator was added. See [this example](https://github.com/faker-ruby/faker/blob/aa31845ed54c25eb2638d716bfddf59771b502aa/lib/faker/music/opera.rb#L68).

Finally, if your pull request affects documentation, please follow the [Guidelines](https://github.com/faker-ruby/faker/blob/master/CONTRIBUTING.md#documentation).

Thanks for contributing to Faker! -->

thdaraujo commented on test/faker/default/test_faker_birthday_in_leap_year.rb:
```diff
@@ -19,24 +19,5 @@ def test_birthday_in_leap_year
-        ::Date.new(@today.year - @max, @today.month, @today.day)
-      end
-    elsif RUBY_VERSION >= '2.7'
-      assert_raise Date::Error do
-        ::Date.new(@today.year - @min, @today.month, @today.day)
-      end
-
-      assert_raise Date::Error do
-        ::Date.new(@today.year - @max, @today.month, @today.day)
-      end
```
Yes, we want to remove the tests for version < 2.7 and we don't need the version checks anymore.

But we still want to keep the tests that check that the error raised is a `Date::Error`, right? I don't think you need to remove them.

thdaraujo commented on test/faker/default/test_faker_id_number.rb:
```diff
@@ -58,17 +58,6 @@ def test_valid_south_african_id_number
-        Date.parse(south_african_id_number_to_date_of_birth_string(sample))
-      end
-    elsif RUBY_VERSION >= '2.7'
-      assert_raises Date::Error do
-        Date.parse(south_african_id_number_to_date_of_birth_string(sample))
-      end
```
same idea here, we still want this test

nickmendezFlatiron replied:
I will fix it and send a new request , thank you for your help @thdaraujo

thdaraujo reviewed (changes requested):
nice, thanks for working on this!

I left a couple of questions regarding the tests we probably still want to keep.

thdaraujo commented:
what do you think of also removing older Ruby versions from the test matrix?
https://github.com/faker-ruby/faker/blob/4ffc714029c1fe22eb0cf091295afa26cb5fa70c/.github/workflows/ruby.yml#L38

nickmendezFlatiron commented:
@thdaraujo I removed the ruby versions older than 2.7 from the test matrix and fixed the 2 errors from my previous pull request. Thank you for your help

nickmendezFlatiron reviewed (commented):
I revised the two test files to include the errors without the version check. I also removed the older ruby versions (  < 2.7) from the test matrix

stefannibrasil commented on test/faker/default/test_faker_id_number.rb:
```diff
@@ -58,23 +58,20 @@ def test_valid_south_african_id_number
     assert_match(/^\d{11}$/, sample)
     assert_match(/(\d)((?!\1)\d)+/, sample)
+
+    assert_raises Date::Error do
```
I believe this new block here is duplicated?

nickmendezFlatiron replied:
Thank you for pointing that out, I have removed it and autocorrected the offenses using rubocop as suggested. Thank you for your assistance

stefannibrasil reviewed (changes requested):
Thank you, @nickmendezFlatiron this is great! I believe there is a duplicated block to be removed. Could you remove it and then we can approve this? Thanks!

stefannibrasil commented:
@nickmendezFlatiron if you run `rubocop -A`, it will fix the errors for you :eyes:

nickmendezFlatiron reviewed (commented):
I have made the suggested changes using rubocop and removed the duplicate code from the test file. Thank you for your help

stefannibrasil reviewed (approved):
Awesome, thank you!

thdaraujo reviewed (approved):
LGTM
